content/en/blog/plugins-early-access.md

```diff
@@ -0,0 +1,157 @@
+
+When Falco loads rules, it reads the `source:` property and tries to match that event source to a loaded source and/or extractor plugin. If a source plugin is loaded that supports that event source, all rules with that source will be evaluated for events coming from that source plugin.
+
+Extractor plugins are handled slightly differently. An extractor plugin optionally provides a set of compatible event sources. When the framework receives an event with an event source that is in the compatible event sources list, fields in expressions/Falco outputs will be extracted from events using the extractor plugin. An extractor plugin can also not name a set of event sources. In this case, all events will be presented to the extractor plugin, regardless of source. In this case, the extractor plugin must detect the format of arbitrary payloads and be able to return NULL/no value when the payload is not supported.
+
+### Configuring Plugins in Falco
+
+Plugins are configured in yaml via new `falco.yaml` properties `plugins` and `load_plugins`. Here's an example:
+
+```yaml
+plugins:
+  - name: cloudtrail
+    library_path: cloudtrail/plugin.so
+    init_config: "..."
+    open_params: "..."
+  - name: json
+    library_path: json/plugin.so
+    init_config: ""
+    open_params: ""
+
+# Optional
+load_plugins: [cloudtrail, http_json]
+```    
+
+The `plugins` property in falco.yaml defines the set of plugins that can be loaded by Falco. It contains a list of plugin definitions. Each plugin definition has:
+
+* `name`: the name of the plugin
+* `library_path`: a path to the plugin shared library
+* `init_config`: A configuration string (typically JSON) that is passed to the plugin when the plugin is initialized.
+* `open_params`: A configuration string that is passed to the plugin when the plugin `open()s` a new stream of events.
+
+By default, every plugin in the plugins list will be loaded. If present, the `load_plugins` property contains specific name entries from `plugins` that are loaded instead.
+
+#### Running Plugins in Falco
+
+For now, when plugin support is enabled, Falco's core support for syscalls is disabled. Additionally, only a single source plugin can be loaded at once. If multiple source plugins are loaded, falco will return an error. Any number of extractor plugins can be loaded at once.
+
+We do plan on relaxing these restrictions to allow running multiple source plugins at once, once we tackle the concurrency, fairness, and resource utilization issues that result from having multiple streams of events being processed by a single falco rules engine.
```

Comment:
We intend to relax these restrictions to allow running multiple source plugins at once. It will happen when we tackle the concurrency, fairness, and resource utilization issues that result from having multiple streams of events being processed by a single Falco rules engine.

Reply:
Sure, updated.